### PR TITLE
v2.0

### LINKED ISSUE #1: grep
`./grep`

### PR DESCRIPTION
**Pull request by leeoocca**
- Integrated a new installation and upgrade cycle.
Very cool uninstall process as well (it cleans trailing spaces and new lines at the end of `.zshrc`).
- Renamed binaries to consistent pattern. Renamed `/src` to `/bin`.
- A `rc` is now called in `.zshrc`
- `/bin` is added to PATH, allowing you to use the helpers wherever you want.
- Small improvements to the helpers. Probably fixed #1.

- Added uninstall confirmation prompt.
- `install.sh` now checks the installation environment and asks for confirmation if something seems off.
- Fixed some hard-coded paths to use the FTH variable instead.

### CODE REVIEW

File: tools/upgrade.sh [hunk 1]
```diff
@@ -1,24 +1,38 @@
...
+function upgrade_prompt {
+	while read -t -k 1 option; do true; done
+	[[ "$option" != ($'\n'|"") ]] && echo
+	echo -n "[42helpers] Would you like to upgrade? [Y/n] "
+	read -r -k 1 option
+	[[ "$option" != $'\n' ]] && echo
+	case "$option" in
+		[yY$'\n'])
+			echo "Upgrading ${bold}42helpers${normal}..."
+			git -C ~/.42helpers pull origin main --no-rebase
```
Comment (leeoocca) [inline tools/upgrade.sh]: `git -C ${FTH}` instead of explicit path

Comment (leeoocca) [inline tools/upgrade.sh]: Are the pull flags the right ones?

File: tools/upgrade.sh [hunk 2]
```diff
@@ -1,24 +1,38 @@
...
+		[nN]) echo "Ok, maybe next time." ;;
+	esac
+}
 
 if [ -d ~/.42helpers ]
 then
-	cd ~/.42helpers
-	git fetch origin
-	if ! git diff --exit-code origin/main..main > /dev/null
+	if ! git -C ~/.42helpers fetch origin &> /dev/null
```
Comment (leeoocca) [inline tools/upgrade.sh]: `git -C ${FTH}` instead of explicit path

File: tools/upgrade.sh [hunk 3]
```diff
@@ -1,24 +1,38 @@
...
 	then
-		echo "Upgrading ${bold}42helpers${normal}..."
-		git pull origin main --no-rebase
-		echo "Done :-)"
+		echo "Couldn't check for upgrades."
+		echo "Check your Internet connection and try again."
+		exit 1
+	fi
+	if ! git -C ~/.42helpers diff --exit-code origin/main..main > /dev/null
```
Comment (leeoocca) [inline tools/upgrade.sh]: `git -C ${FTH}` instead of explicit path

Comment (leeoocca) [reply tools/upgrade.sh]: Fixed in 80d7ed4

File: tools/upgrade.sh (hunk 2 above)
Comment (leeoocca) [reply tools/upgrade.sh]: Fixed in 80d7ed4

File: tools/upgrade.sh (hunk 1 above)
Comment (leeoocca) [reply tools/upgrade.sh]: Fixed in 80d7ed4

Comment (leeoocca) [reply tools/upgrade.sh]: Fixed in 80d7ed4

**Review by UmbertoSavoia — APPROVED** [review]

**Pull request merged**